compiler/rustc_trait_selection/src/traits/mod.rs

```diff
@@ -219,6 +220,7 @@ fn do_normalize_predicates<'tcx>(
     cause: ObligationCause<'tcx>,
     elaborated_env: ty::ParamEnv<'tcx>,
     predicates: Vec<ty::Predicate<'tcx>>,
```

Comment:
From the use site, should we just take `&mut`?

Reply:
Yea, as mentioned above, fully_normalize requires taking ownership of the predicats. I could not think of a good way to deal with this for a while.